src/systems/systems/src/air_conditioning/acs_controller.rs

```diff
@@ -0,0 +1,1808 @@
+use crate::{
+    shared::{pid::PidController, CabinAltitude, EngineCorrectedN1, LgciuWeightOnWheels},
+    simulation::{
+        InitContext, Read, Reader, SimulationElement, SimulationElementVisitor, SimulatorReader,
+        SimulatorWriter, UpdateContext, VariableIdentifier, Write, Writer,
+    },
+};
+
+use super::{
+    AirConditioningSystemOverhead, DuctTemperature, FlowControlValveSignal, PackFlow, PackFlowValve,
+};
+
+use std::{collections::HashMap, time::Duration};
+
+use uom::si::{
+    f64::*,
+    length::foot,
+    mass_rate::kilogram_per_second,
+    ratio::{percent, ratio},
+    thermodynamic_temperature::{degree_celsius, kelvin},
+    velocity::knot,
+};
+
+pub(super) struct ACSController {
+    aircraft_state: AcStateManager,
+    cabin_zone_ids: Vec<&'static str>,
+    zone_controller: Vec<ZoneController>,
+    pack_flow_controller: PackFlowController,
+}
+
+impl ACSController {
+    pub fn new(context: &mut InitContext, cabin_zone_ids: Vec<&'static str>) -> Self {
+        let zone_controller = cabin_zone_ids
+            .iter()
+            .map(|id| ZoneController::new(context, id))
+            .collect::<Vec<ZoneController>>();
+        Self {
+            aircraft_state: AcStateManager::new(),
+            cabin_zone_ids,
+            zone_controller,
+            pack_flow_controller: PackFlowController::new(context),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        acs_overhead: &AirConditioningSystemOverhead,
+        pack_flow_valve: &[PackFlowValve; 2],
+        engines: [&impl EngineCorrectedN1; 2],
+        pressurization: &impl CabinAltitude,
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+    ) {
+        self.aircraft_state = self.aircraft_state.update(context, engines, lgciu);
+        for zone in self.zone_controller.iter_mut() {
+            zone.update(
+                context,
+                acs_overhead,
+                &self.pack_flow_controller,
+                pressurization,
+            )
+        }
+        self.pack_flow_controller.update(
+            &self.aircraft_state,
+            acs_overhead,
+            engines,
+            pressurization,
+            pack_flow_valve,
+        );
+    }
+}
+
+impl DuctTemperature for ACSController {
+    fn duct_demand_temperature(&self) -> HashMap<&'static str, ThermodynamicTemperature> {
+        let mut duct_temperature: HashMap<&str, ThermodynamicTemperature> = HashMap::new();
+        for (id, zone) in self.cabin_zone_ids.iter().zip(&self.zone_controller) {
+            duct_temperature.insert(id, zone.duct_demand_temperature()[zone.zone_id()]);
+        }
+        duct_temperature
+    }
+}
+
+impl PackFlow for ACSController {
+    fn pack_flow(&self) -> MassRate {
+        self.pack_flow_controller.pack_flow()
+    }
+}
+
+impl FlowControlValveSignal for ACSController {
+    fn should_open_fcv(&self) -> [bool; 2] {
+        self.pack_flow_controller.should_open_fcv()
+    }
+}
```

Comment:
I've now wrapped this in `ControllerSignal<S>`. I wanted to integrate/merge it with the `PackFlowValveController` in the Bleed system, but after spending a good amount of time looking at it I didn't find an easy straight forward way to do this within this PR without increasing the scope considerably. 

At the moment both systems work independently from each other. I think we can leave this for a future PR since the packs also need to be modelled at some point. What do you think @davidwalschots and @BlueberryKing ?

Reply:
My `PackFlowValveController` is more of a placeholder than anything else. Feel free to remove it and emit the signal from the ACSC instead. It's also perfectly fine to leave the systems entirely separate for now.